src/main/java/io/odpf/firehose/sink/objectstorage/writer/local/LocalFileChecker.java

```diff
@@ -1,52 +1,90 @@
-    }
+        toBeRotated.forEach((key, writer) -> {
+            String filePath = writer.getFullPath();
+            Partition partition = localStorage.getPartitionFactory().fromPartitionPath(key);
+            String dateTimeTag = partition.getDatetimePathWithoutPrefix();
+            String topic = partition.getTopic();
+            try {
+                Instant fileClosingStartTime = clock.now();
+                writer.close();
+
+                //getting size of file from file instead of writer, because there is possibility of data compression on the writer
+                long fileSize = localStorage.getFileSize(filePath);
+                FileMeta fileMeta = new FileMeta(filePath, writer.getRecordCount(), fileSize, partition);
+
+                toBeFlushedToRemotePaths.add(fileMeta);
+                LOGGER.info("Closing Local File " + filePath);
+
+                instrumentation.incrementCounterWithTags(SINK_OBJECT_STORAGE_LOCAL_FILE_CLOSE_TOTAL,
+                        SUCCESS_TAG,
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
+
+                instrumentation.captureDurationSinceWithTags(SINK_OBJECT_STORAGE_LOCAL_FILE_CLOSING_TIME_MILLISECONDS, fileClosingStartTime,
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
+
+                instrumentation.captureCountWithTags(SINK_OBJECT_STORAGE_RECORD_PROCESSED_TOTAL, fileMeta.getRecordCount(),
+                        tag(SINK_OBJECT_STORAGE_SCOPE_TAG, SINK_OBJECT_STORAGE_SCOPE_FILE_CLOSE),
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
 
-    private Boolean shouldRotate(LocalFileWriter writer) {
-        return policies.stream().reduce(false,
-                (accumulated, writerPolicy) -> accumulated || writerPolicy.shouldRotate(writer), (left, right) -> left || right);
+                instrumentation.captureCountWithTags(SINK_OBJECT_STORAGE_LOCAL_FILE_SIZE_BYTES, fileMeta.getFileSizeBytes(),
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
+            } catch (IOException e) {
+                e.printStackTrace();
+                instrumentation.captureCountWithTags(SINK_OBJECT_STORAGE_RECORD_PROCESSING_FAILED_TOTAL, writer.getRecordCount(),
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
+                instrumentation.incrementCounterWithTags(SINK_OBJECT_STORAGE_LOCAL_FILE_CLOSE_TOTAL,
+                        FAILURE_TAG,
+                        tag(SINK_OBJECT_STORAGE_TOPIC_TAG, topic),
+                        tag(SINK_OBJECT_STORAGE_PARTITION_TAG, dateTimeTag));
```

Comment:
as we discussed earlier, tag value cannot be unbounded. We might need to remove partition as a tag from metrics.